azurerm/resource_arm_key_vault_key.go

```diff
@@ -174,6 +174,11 @@ func resourceArmKeyVaultKeyRead(d *schema.ResourceData, meta interface{}) error
 
 	resp, err := client.GetKey(ctx, id.KeyVaultBaseUrl, id.Name, "")
 	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
```

Comment:
It'd be great to log that we're removing the key from state because it wasn't found.

Reply:
done